adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt

```diff
@@ -8,26 +8,56 @@ class VersionData {
     val minor: Int
     val security: Int
     val pre: String?
-    val adopt_build_number: Int
+    val adopt_build_number: Int?
 
     @Schema(example = "11.0.0+28")
-    val semver: String
+    val semver: String?
 
     @Schema(example = "11.0.4+10-201907081820")
     val openjdk_version: String
     val build: Int
     val optional: String?
 
-    constructor(major: Int, minor: Int, security: Int, pre: String?, adopt_build_number: Int, semver: String, build: Int, optional: String?, openjdk_version: String) {
+    constructor(major: Int, minor: Int, security: Int, pre: String?, adopt_build_number: Int?, build: Int, optional: String?, openjdk_version: String) {
+        this.major = major
         this.minor = minor
         this.security = security
-        this.pre = pre
-        this.adopt_build_number = adopt_build_number
-        this.major = major
-        this.semver = semver
+        this.pre = if (pre?.isNotEmpty() == true) pre else null
+        this.adopt_build_number = if (adopt_build_number != null && adopt_build_number != 0) adopt_build_number else null
         this.build = build
-        this.optional = optional
+        this.optional = if (optional?.isNotEmpty() == true) optional else null
         this.openjdk_version = openjdk_version
+        this.semver = formSemver()
+    }
+
+
+    // i.e 11.0.1+11.1
+    fun formSemver(): String? {
+        var semver = major.toString() + "." + minor + "." + security
```

Comment:
Is it really `security`? semver.org calls this `patch`, JEP 223 calls this `update` and the 4th number `patch`.

Pick one and say who's scheme you're following and say it's bullshit ;-)

Reply:
actually 223 calls this a security, can leave it